### DISCUSSION TITLE
AutoGluonTimeseries: Training multiple time series/items at once

### DISCUSSION THREAD

**Discussion by mputra533** [autogluon/autogluon#4117] [Q&A]
Hi!

A few questions about training multiple items in one big dataframe (separated by item_id)
How does training for multiple time series/item work? 
Does the time_limit apply to each model for all items or each model for one item?
Documentation states "**AutoGluon generates forecasts for each time series individually, without modelling interactions between different items"** Wondering why there's only one best model and validation score for all of the multiple items that are trained as one from the leader board?

I did an experiment where I have 4 normal items and one dummy item with all 0 values. I trained all 5 items at once from one dataframe, separated by the unique item_id's. I expect the item with all 0 values to have prediction of all 0's but it wasn't. The prediction seems to be influenced by the other 4 normal items? I sort of followed the trend from the other 4. But, when I train the item with all 0 by itself, it predicted all 0's - which is what I expected.

**Comment by shchur** [comment]
Hi @mputra533, thank you for the question! 

All forecasting models in AutoGluon are univariate, which means that each model generates predictions for each time series in "isolation", without looking at other time series. In other words, *after we trained the model*, if we change history of item A this won't affect the forecast of item B.

However, some models (such as DeepAR or WeightedEnsemble) are trained **globally**. This means, during training we are looking for a single model that on average performs best for **all** items. In case of ensemble, this means that the ensemble weights are shared across all items. This means that in your example:
- when you trained only on the time series of all zeros, the best model was probably Naive that produces the best forecast for the all-zero item.
- when you included 4 "normal" time series, the ensemble allocated non-zero weight to other models (e.g., DeepAR) that perform well for the "normal" items. For this reason, the forecast also changed for the all-zero item.

**Reply by mputra533** [reply]
Hi @shchur thank you for your reply. In other words, are you recommending to exclude DeepAR and ensemble models for all 5 to work individually?

**Reply by shchur** [reply]
If the 5 time series that you want to forecast are very different, and you expect that very different models will be best at forecasting them, I would recommend to 
- create a separate predictor for each time series
- exclude expensive global models like DeepAR, TFT, PatchTST
- keep ensemble enabled
This will likely be slower than training on all series simultaneously, but might result in better accuracy.

**Reply by mputra533** [reply]
Thank you for the recommendation, to clarify:
Create separate predictor then call `fit()` separately on the 5 individual dataframes, and then call `predict()` with the 5 dataframes all together?

I'm still trying to understand this statement 

> "AutoGluon generates forecasts for each time series individually, without modelling interactions between different items"

If I have 5 different items, why do we need to generate a separate predictor and call `fit()` separately? Wouldn't this take significantly longer compared to creating one for all?

**Reply by shchur** [reply]
Global models such as the WeightedEnsemble train a single model that "works well on average" across all the time series. If the time series that you are trying to forecast are very different, it might make sense to train different models to forecast them. This, of course, will be slower and more expensive.

This is question that cannot really be answered in general, one must run experiments on their data to see which of the two strategies works better:
1. Train a single model for each time series (slower)
2. Train one global model for all time series (faster)

Approach 2 often works better if the time series are somehow similar / related (e.g., sales for different similar products). If the time series are very different from each other, then approach 1 might work better.
